packages/api/src/providers/__tests__/oauth1-deregister.test.ts

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import * as dotenv from "dotenv";
+dotenv.config({ path: ".env.test" });
+// Keep dotenv import and config before everything else
+import * as updateProviderDataFile from "../../command/connected-user/save-connected-user";
+import * as getUserTokenFile from "../../command/cx-user/get-user-token";
+import * as saveUserTokenFile from "../../command/cx-user/save-user-token";
+import {
+  TestGarmin,
+  anotherUserTokenMocked,
+  testUser,
+  testUserModified,
+  thirdTestUser,
+  userTokenMocked,
+  userTokenModified,
+} from "./deregister-input";
+
+let getUserTokenByUATMock: jest.SpyInstance;
+let saveUserTokenMock: jest.SpyInstance;
+let updateProviderDataMock: jest.SpyInstance;
+
+beforeEach(() => {
+  jest.restoreAllMocks();
+  getUserTokenByUATMock = jest.spyOn(getUserTokenFile, "getUserTokenByUAT");
+  saveUserTokenMock = jest.spyOn(saveUserTokenFile, "saveUserToken");
+  updateProviderDataMock = jest.spyOn(updateProviderDataFile, "updateProviderData");
+});
+
+const testGarmin = new TestGarmin();
```

Comment:
Do we want to call this `garmin` b/c we're testing Garmin implementation? If I understand correctly, we're testing `OAuth1DefaultImpl`, no? Maybe we should call these `oauth1` and related instead?

Reply:
Since we're using a Garmin class to instantiate the testGarmin, I think it's best to keep the name. Oauth1 is a private readonly resident inside Garmin, which we cannot use directly to invoke its methods.

Reply:
> Since we're using a Garmin class to instantiate the testGarmin

That's what I mean, why are we calling that class Garmin related? Its not testing Garmin code in anyway, its testing the code on `OAuth1DefaultImpl`, isn't? If we were testing the Garmin class, (1) we wouldn't need this implementation and (2) we would only validate it calls OAuth's implementation of `deregister` with the right parameters.

In fact, we don't even need a different implementation of OAuth1, do we? Couldn't we just instantiate `OAuth1DefaultImpl` and tests its functions?

Reply:
Yes, very good point. I tried to instantiate `OAuth1DefaultImpl` before, but something was going wrong, whereas Garmin worked right away. But now that I tried it out again, it's also working and is 100% a better implementation.